### PR TITLE
Adding Firebase storage and display of location

### PR DESCRIPTION
**Pull request by macintosh1398**
This PR give the user the ability to select a location for a particular deadline and the location name and coordinates (GeoPoint) will be stored on Firebase. When the user looks at the deadline details, the location name will appear (if there is one). Some UI improvements/icon addition were also made (main screen and deadline details).

### CODE REVIEW

**Comment by macintosh1398** [comment]
New test was added for serialization to have full coverage. But in the other hand, I had to remove AddDeadlineActivity tests in order to solve the testing bug of mapbox. This results in a big drop of code coverage. Needs to be solved.

File: app/src/main/java/com/github/multimatum_team/multimatum/activity/MainActivity.kt [hunk 1]
```diff
@@ -23,6 +23,7 @@ import com.github.multimatum_team.multimatum.repository.PdfRepository
 import com.github.multimatum_team.multimatum.util.DeadlineNotification
 import com.github.multimatum_team.multimatum.viewmodel.DeadlineListViewModel
 import com.github.multimatum_team.multimatum.viewmodel.GroupViewModel
+import com.google.firebase.firestore.FirebaseFirestore
```
Comment (Matathiasq13824) [inline app/src/main/java/com/github/multimatum_team/multimatum/activity/MainActivity.kt:26]: Does this import is still useful ?

**Review by Matathiasq13824 — APPROVED** [review]
Great job by finding a way to implement this without removing the AddDeadline tests. The coverage is still low but it can be solved next sprint. I have find only a little problem that I have commented but It is too small to not approve now

File: app/src/main/java/com/github/multimatum_team/multimatum/activity/DeadlineDetailsActivity.kt [hunk 2]
```diff
@@ -318,6 +321,8 @@ class DeadlineDetailsActivity : AppCompatActivity() {
             dateView.text =
                 getString(R.string.DueTheXatX, dateTime.toLocalDate(), dateTime.toLocalTime())
             titleView.text = SpannableStringBuilder(title)
+            if (deadline.locationName != null) locationView.text = deadline.locationName
+            locationView.text = deadline.locationName
```
Comment (ValentinAebi) [inline app/src/main/java/com/github/multimatum_team/multimatum/activity/DeadlineDetailsActivity.kt:325]: I think there is a problem here...

**Review by ValentinAebi — CHANGES_REQUESTED** [review]
Congrats on fixing the bug! 🥇 
Which solution worked in the end?

File: app/src/main/java/com/github/multimatum_team/multimatum/activity/MainActivity.kt (hunk 1 above)
Comment (macintosh1398) [reply app/src/main/java/com/github/multimatum_team/multimatum/activity/MainActivity.kt:26]: No, removed

File: app/src/main/java/com/github/multimatum_team/multimatum/activity/DeadlineDetailsActivity.kt (hunk 2 above)
Comment (macintosh1398) [reply app/src/main/java/com/github/multimatum_team/multimatum/activity/DeadlineDetailsActivity.kt:325]: Oh yes your're right. Thanks. About Mapbox, I created a new activity called SearchLocation that contains the mapbox view. This activity is lauched using the button on the addDeadlineActivity and the location result is communicated via intent. This is done using the StartActivityForResult() function.

Comment (macintosh1398) [reply app/src/main/java/com/github/multimatum_team/multimatum/activity/DeadlineDetailsActivity.kt:325]: So it this "solution" give us the ability to execute addDeadlineTests without any mapbox stuff in it.

Comment (ValentinAebi) [reply app/src/main/java/com/github/multimatum_team/multimatum/activity/DeadlineDetailsActivity.kt]: OK nice. I am just wondering where the coverage drop comes from... From CodeClimate I see that `SearchLocationActivity` has 0% coverage, is that correct?

**Review by ValentinAebi — APPROVED** [review]

**Pull request merged**